pymc/distributions/distribution.py

```diff
@@ -601,6 +661,20 @@ def update(self, node: Node):
         return updates
 
 
+def dist_moment(rv, *args):
+    node = rv.owner
+    op = node.op
+    rv_out_idx = node.outputs.index(rv)
+
+    fgraph = op.fgraph.clone()
+    replace_moments = MomentRewrite()
+    replace_moments.rewrite(fgraph)
+    # Replace dummy inner inputs by outer inputs
+    fgraph.replace_all(tuple(zip(op.inner_inputs, node.inputs)), import_missing=True)
```

Comment:
Can use `args` once you dispatch this to the default case, as we won't be filtering the RVs anymore.
I suggest calling it `*inputs`

```suggestion
    fgraph.replace_all(tuple(zip(op.inner_inputs, args)), import_missing=True)
```